Structure_Engine/Objects/EqualityComparers/SectionPropertyComparer.cs

```diff
@@ -0,0 +1,65 @@
+﻿/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2023, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+
+using BH.oM.Structure.SectionProperties;
+using System.Collections.Generic;
+using System;
+
+namespace BH.Engine.Structure
+{
+    public class SectionPropertyComparer : IEqualityComparer<ISectionProperty>
```

Comment:
Is there a way of testing the comparers (for section and surface properties)? I think most knowledge about this concept is currently gone on leave together with @IsakNaslundBh, so I would not be overly confident with reviewing/approving this bit tbh.

Reply:
This is the one bit I was really no confident on - I think the reason we don't have `SectionPropertyComparer` and `SurfacePropertyComparer` is because how many levels deep you have to compare (and then you'd have to have a `MaterialFragmentComparer` too). 

I only stuck these in because they're needed for `IsMergeableProperties` but I wonder if we can just return `false` if that method is absolutely required.

Reply:
No clue 🙈 Let's try to chat it through offline, maybe we can come up with something that would make us more confident

Reply:
As per discussion, I will remove the two methods for `Pile` and `PadFoundation` and we will rely on the `IElement1D` and `IElement2D` methods in the `Spatial_Engine`. The onus will be on the user to group similar `PadFoundation` and `Pile` (or use the `Filter` methods in `Data_Engine`.